src/SignalR/clients/csharp/Client.Core/src/Microsoft.AspNetCore.SignalR.Client.Core.csproj

```diff
@@ -13,6 +13,7 @@
     <Compile Include="$(SignalRSharedSourceRoot)PipeWriterStream.cs" Link="PipeWriterStream.cs" />
     <Compile Include="$(SignalRSharedSourceRoot)ReflectionHelper.cs" Link="ReflectionHelper.cs" />
     <Compile Include="$(SignalRSharedSourceRoot)TimerAwaitable.cs" Link="Internal\TimerAwaitable.cs" />
+    <Compile Include="$(SignalRSharedSourceRoot)ValueTaskExtensions.cs" Link="Internal\ValueTaskExtensions.cs" />
```

Comment:
IMO instead of placing this code under SignalR's shared source directory, it should be under the repos shared source directory: `src/Shared` -> https://github.com/aspnet/AspNetCore/tree/master/src/Shared

That way Kestrel and SignalR can use the same copy of code. There is no overhead of creating a source package because we mono repo now.

Reply:
@davidfowl

Reply:
Do it in another PR. I don't care that much.